File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/InputsAndResultsTestString.js
print("getString")
if(typeof getString !== 'undefined')
	throw "getString already defined!"
getString = putString
getStringType = typeof getString
print(getStringType)
print(getString)
if(getStringType !== 'string')
	throw "getString not string!"

print("getStringWithNull")
if(typeof getStringWithNull !== 'undefined')
	throw "getStringWithNull already defined!"
getStringWithNull = putStringWithNull
getStringWithNullType = typeof getStringWithNull
print(getStringWithNullType)
print(getStringWithNull)
if(getStringWithNull !== null)
	throw "getStringWithNull not null!"

print("getStringVector")
if(typeof getStringVector !== 'undefined')
	throw "getStringVector already defined!"
getStringVector = putStringVector
getStringVectorType = typeof getStringVector[0]
print(getStringVectorType)
print(getStringVector)
if(getStringVectorType !== 'string')
	throw "getStringVector not string!"

print("getStringVectorWithNull")
if(typeof getStringVectorWithNull !== 'undefined')
	throw "getStringVectorWithNull already defined!"
getStringVectorWithNull = putStringVectorWithNull
getStringVectorWithNullType = typeof getStringVectorWithNull[0]
print(getStringVectorWithNullType)
print(getStringVectorWithNull)
if(getStringVectorWithNullType !== 'string')
	throw "getStringVectorWithNull not string!"
if(getStringVectorWithNull[1] !== null)
	throw "getStringVectorWithNull[2] not null!"

print("getStringVectorAsList")
if(typeof getStringVectorAsList !== 'undefined')
	throw "getStringVectorAsList already defined!"
getStringVectorAsList = putStringVectorAsList
getStringVectorAsListType = typeof getStringVectorAsList[0]
print(getStringVectorAsListType)
print(getStringVectorAsList)
if(getStringVectorAsListType !== 'string')
	throw "getStringVectorAsList not string!"

print("getStringVectorAsListWithNull")
if(typeof getStringVectorAsListWithNull !== 'undefined')
	throw "getStringVectorAsListWithNull already defined!"
getStringVectorAsListWithNull = putStringVectorAsListWithNull
getStringVectorAsListWithNullType = typeof getStringVectorAsListWithNull[0]
print(getStringVectorAsListWithNullType)
print(getStringVectorAsListWithNull)
if(getStringVectorAsListWithNullType !== 'string')
	throw "getStringVectorAsListWithNull not string!"
if(getStringVectorAsListWithNull[1] !== null)
	throw "getStringVectorAsListWithNull[1] not null!"

print("getStringMatrix")
if(typeof getStringMatrix !== 'undefined')
	throw "getStringMatrix already defined!"
getStringMatrix = putStringMatrix
getStringMatrixType = typeof getStringMatrix[0][0]
print(getStringMatrixType)
print(getStringMatrix)
if(getStringMatrixType !== 'string')
	throw "getStringMatrix not string!"

print("getStringMatrixWithNull")
if(typeof getStringMatrixWithNull !== 'undefined')
	throw "getStringMatrixWithNull already defined!"
getStringMatrixWithNull = putStringMatrixWithNull
getStringMatrixWithNullType = typeof getStringMatrixWithNull[0][1]
print(getStringMatrixWithNullType)
print(getStringMatrixWithNull)
if(getStringMatrixWithNullType !== 'string')
	throw "getStringMatrixWithNull not string!"
if(getStringMatrixWithNull[0][0] !== null)
	throw "getStringMatrixWithNull[0][0] not null!"
if(getStringMatrixWithNull[1][1] !== null)
	throw "getStringMatrixWithNull[1][1] not null!"
if(getStringMatrixWithNull[2][2] !== null)
	throw "getStringMatrixWithNull[2][2] not null!"

print("getStringMatrixAsList")
if(typeof getStringMatrixAsList !== 'undefined')
	throw "getStringMatrixAsList already defined!"
getStringMatrixAsList = putStringMatrixAsList
getStringMatrixAsListType = typeof getStringMatrixAsList[0][0]
print(getStringMatrixAsListType)
print(getStringMatrixAsList)
if(getStringMatrixAsListType !== 'string')
	throw "getStringMatrixAsList not string!"

print("getStringMatrixAsListWithNull")
if(typeof getStringMatrixAsListWithNull !== 'undefined')
	throw "getStringMatrixAsListWithNull already defined!"
getStringMatrixAsListWithNull = putStringMatrixAsListWithNull
getStringMatrixAsListWithNullType = typeof getStringMatrixAsListWithNull[0][1]
print(getStringMatrixAsListWithNullType)
print(getStringMatrixAsListWithNull)
if(getStringMatrixAsListWithNullType !== 'string')
	throw "getStringMatrixAsListWithNull not string!"
if(getStringMatrixAsListWithNull[0][0] !== null)
	throw "getStringMatrixAsListWithNull[0][0] not null!"
if(getStringMatrixAsListWithNull[1][1] !== null)
	throw "getStringMatrixAsListWithNull[1][1] not null!"
if(getStringMatrixAsListWithNull[2][2] !== null)
	throw "getStringMatrixAsListWithNull[2][2] not null!"

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/InputsAndResultsTestString.rb
puts("getString")
unless (defined?(getString)).nil? || getString.nil?
	raise "getString already defined!"
end
getString = putString
getStringType = getString.class
puts(getStringType)
puts(getString)
if(getStringType != String)
	raise "getString not String!"
end

puts("getStringWithNull")
unless (defined?(getStringWithNull)).nil? || getStringWithNull.nil?
	raise "getStringWithNull already defined!"
end
getStringWithNull = putStringWithNull
getStringWithNullType = getStringWithNull.class
puts(getStringWithNullType)
puts(getStringWithNull)
unless(getStringWithNull.nil?)
	raise "getStringWithNull not null!"
end

puts("getStringVector")
unless (defined?(getStringVector)).nil? || getStringVector.nil?
	raise "getStringVector already defined!"
end
getStringVector = putStringVector
getStringVectorType = getStringVector[0].class
puts(getStringVectorType)
puts(getStringVector)
if(getStringVectorType != String)
	raise "getStringVector not String!"
end

puts("getStringVectorWithNull")
unless (defined?(getStringVectorWithNull)).nil? || getStringVectorWithNull.nil?
	raise "getStringVectorWithNull already defined!"
end
getStringVectorWithNull = putStringVectorWithNull
getStringVectorWithNullType = getStringVectorWithNull[0].class
puts(getStringVectorWithNullType)
puts(getStringVectorWithNull)
if(getStringVectorWithNullType != String)
	raise "getStringVectorWithNull not String!"
end
unless(getStringVectorWithNull[1].nil?)
	raise "getStringVectorWithNull[1] not null!"
end

puts("getStringVectorAsList")
unless (defined?(getStringVectorAsList)).nil? || getStringVectorAsList.nil?
	raise "getStringVectorAsList already defined!"
end
getStringVectorAsList = putStringVectorAsList
getStringVectorAsListType = getStringVectorAsList[0].class
puts(getStringVectorAsListType)
puts(getStringVectorAsList)
if(getStringVectorAsListType != String)
	raise "getStringVectorAsList not String!"
end

puts("getStringVectorAsListWithNull")
unless (defined?(getStringVectorAsListWithNull)).nil? || getStringVectorAsListWithNull.nil?
	raise "getStringVectorAsListWithNull already defined!"
end
getStringVectorAsListWithNull = putStringVectorAsListWithNull
getStringVectorAsListWithNullType = getStringVectorAsListWithNull[0].class
puts(getStringVectorAsListWithNullType)
puts(getStringVectorAsListWithNull)
if(getStringVectorAsListWithNullType != String)
	raise "getStringVectorAsListWithNull not String!"
end
unless(getStringVectorAsListWithNull[1].nil?)
	raise "getStringVectorAsListWithNull[1] not null!"
end

puts("getStringMatrix")
unless (defined?(getStringMatrix)).nil? || getStringMatrix.nil?
	raise "getStringMatrix already defined!"
end
getStringMatrix = putStringMatrix
getStringMatrixType = getStringMatrix[0][0].class
puts(getStringMatrixType)
puts(getStringMatrix)
if(getStringMatrixType != String)
	raise "getStringMatrix not String!"
end

puts("getStringMatrixWithNull")
unless (defined?(getStringMatrixWithNull)).nil? || getStringMatrixWithNull.nil?
	raise "getStringMatrixWithNull already defined!"
end
getStringMatrixWithNull = putStringMatrixWithNull
getStringMatrixWithNullType = getStringMatrixWithNull[0][1].class
puts(getStringMatrixWithNullType)
puts(getStringMatrixWithNull)
if(getStringMatrixWithNullType != String)
	raise "getStringMatrixWithNull not String!"
end
unless(getStringMatrixWithNull[0][0].nil?)
	raise "getStringMatrixWithNull[0][0] not null!"
end
unless(getStringMatrixWithNull[1][1].nil?)
	raise "getStringMatrixWithNull[1][1] not null!"
end
unless(getStringMatrixWithNull[2][2].nil?)
	raise "getStringMatrixWithNull[2][2] not null!"
end

puts("getStringMatrixAsList")
unless (defined?(getStringMatrixAsList)).nil? || getStringMatrixAsList.nil?
	raise "getStringMatrixAsList already defined!"
end
getStringMatrixAsList = putStringMatrixAsList
getStringMatrixAsListType = getStringMatrixAsList[0][0].class
puts(getStringMatrixAsListType)
puts(getStringMatrixAsList)
if(getStringMatrixAsListType != String)
	raise "getStringMatrixAsList not String!"
end

puts("getStringMatrixAsListWithNull")
unless (defined?(getStringMatrixAsListWithNull)).nil? || getStringMatrixAsListWithNull.nil?
	raise "getStringMatrixAsListWithNull already defined!"
end
getStringMatrixAsListWithNull = putStringMatrixAsListWithNull
getStringMatrixAsListWithNullType = getStringMatrixAsListWithNull[0][1].class
puts(getStringMatrixAsListWithNullType)
puts(getStringMatrixAsListWithNull)
if(getStringMatrixAsListWithNullType != String)
	raise "getStringMatrixAsListWithNull not String!"
end
unless(getStringMatrixAsListWithNull[0][0].nil?)
	raise "getStringMatrixAsListWithNull[0][0] not null!"
end
unless(getStringMatrixAsListWithNull[1][1].nil?)
	raise "getStringMatrixAsListWithNull[1][1] not null!"
end
unless(getStringMatrixAsListWithNull[2][2].nil?)
	raise "getStringMatrixAsListWithNull[2][2] not null!"
end
